app/controllers/items_controller.rb

```diff
@@ -2,17 +2,14 @@ class ItemsController < ApplicationController
   include Pagy::Backend
 
   def index
-    item_scope = Item.listed_publicly.includes(:checked_out_exclusive_loan)
+    item_scope = Item.listed_publicly.includes(:borrow_policy, :active_holds, :checked_out_exclusive_loan, :categories)
 
-    if params[:category]
-      @category = CategoryNode.where(id: params[:category]).first
-      redirect_to(items_path, error: "Category not found") && return unless @category
+    item_scope = filter_by_category(item_scope) if filter_params[:category].present?
+    item_scope = filter_by_query(item_scope) if filter_params[:query].present?
```

Comment:
This reads nicely and does a good job of extracting the filtering logic into other methods.